source/javascripts/controllers/_ProvProfileController.js.erb

```diff
@@ -163,28 +192,149 @@ angular.module("BitriseWorkflowEditor").controller("ProvProfileController", func
+			viewModel.uploadProvProfilesProgress.reset();
+		}
+
+		$q.all([
+			validateTestDevices(provProfile),
+			validateCertificates(provProfile)
+		]).then(function() {
+
+			if (viewModel.provProfileConfigs[provProfile.databaseID].validationStatuses.certificateValidationsStatus == "error" || viewModel.provProfileConfigs[provProfile.databaseID].validationStatuses.deviceValidation == "error") {
+				viewModel.provProfileConfigs[provProfile.databaseID].validationStatuses.globalValidationStatus = "error";
+			}
+			else {
+				viewModel.provProfileConfigs[provProfile.databaseID].validationStatuses.globalValidationStatus = "success";
+			}
+
+			viewModel.provProfileConfigs[provProfile.databaseID].validationProgress.success();
+		}, function(error) {
+			viewModel.provProfileConfigs[provProfile.databaseID].validationProgress.error(error);
+		});
+	}
+
+	viewModel.capabilityDescriptionsOfProvProfile = function() {
+		return _.compact(_.uniq(_.map(viewModel.selectedProvProfile.capabilityKeys, function(aCapabilityKey) {
+			return ProvProfile.descriptionForCapabilityKey(aCapabilityKey);
+		})));
+	};
+
+	function validateTestDevices(provProfile) {
+		viewModel.validateTestDevicesProgress.start("<%= data[:strings][:code_signing][:prov_profile][:validate_test_devices][:in_progress] %>");
+
+		return requestService.fetchAllTestDevicesOfRepository().then(function(registeredTestDevices) {
+			_.each(provProfile.provisionedDevices, function(aProvisionedDevice) {
+				var registeredTestDeviceOnBitrise = _.find(registeredTestDevices.devices, function(aDevice) {
+					return aDevice.deviceIdentifier == aProvisionedDevice.udid;
+				});
+
+				if (registeredTestDeviceOnBitrise) {
+					aProvisionedDevice.statusCSSclass = "success";
+					aProvisionedDevice.registeredBy = registeredTestDeviceOnBitrise.owner.username;
+				}
+				else {
+					aProvisionedDevice.statusCSSclass = "error";
+					aProvisionedDevice.registeredBy = undefined;
+				}
+			});
+
+			var areAllValidationsSuccessfull = !_.any(provProfile.provisionedDevices, {
+				statusCSSclass: "error"
+			});
+
+			viewModel.provProfileConfigs[provProfile.databaseID].validationStatuses.deviceValidationStatus = areAllValidationsSuccessfull ? "success" : "error";
+
+			viewModel.validateTestDevicesProgress.success();
+		}, function(error) {
+			viewModel.validateTestDevicesProgress.error(error);
+		});
+	}
+
+	function validateCertificates(provProfile) {
+		viewModel.validateCertificatesProgress.start("<%= data[:strings][:code_signing][:prov_profile][:validate_certificates][:in_progress] %>");
+
+		if (certificateService.progress.isInProgress) {
+			$timeout(function() {
+				validateCertificates(provProfile);
+			}, validateCertificatesReloadIntervalInSeconds * 1000);
+		}
+		else {
+			_.each(provProfile.developerCertificateIdentities, function(anIdentity) {
+				if (certificateService.progress.isError) {
+					anIdentity.statusCSSclass = "error";
+
+					return false;
```

Comment:
You don't need to return false - don't need to return anything, because _.each does not care about the return value.